minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/configuration/notifiers/RestChangeNotifier.java

```diff
@@ -227,15 +235,27 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
                     }
                 }
                 setConfigFile(configBuilder.substring(0,configBuilder.length()-1));
-                notifyListeners();
-                writeOutput(response, POST_TEXT, 200);
+                ListenerHandleResult[] listenerHandleResults = notifyListeners();
+
+                writeOutput(response, getPostText(listenerHandleResults), 200);
```

Comment:
Good call, I changed it to return the text status of the listeners but didn't change the status code. I can't tell whether the listener failed due to client or server error so I will just return 200 if all succeeded but 500 if any failed.